spark/src/main/scala/io/delta/sharing/spark/util/RetryUtils.scala

```
@@ -39,6 +39,8 @@ private[sharing] object RetryUtils extends Logging {
           logWarning(s"Sleeping $sleepMs ms to retry because of error: ${e.getMessage}", e)
           sleeper(sleepMs)
           sleepMs *= 2
+        case e: Exception =>
+          logError(s"Not retrying delta sharing rpc on error: ${e.getMessage}", e)
```

I think this is problematic.... we need to throw the exception.